File: inst/examples/deckgl-api-reference/path-layer.R
## @knitr path-layer
sample_data <- paste0(
  "https://raw.githubusercontent.com/",
  "uber-common/deck.gl-data/",
  "master/website/bart-lines.json"
)

properties <- list(
  pickable = TRUE,
  widthScale = 20,
  widthMinPixels = 2,
  getPath = ~path,
  getColor = ~color,
  getWidth = 5,
  getTooltip = ~name
)

deck <- deckgl(pitch = 25, zoom = 10.5) %>%
  add_path_layer(data = sample_data, properties = properties) %>%
  add_basemap() %>%
  add_control("Path Layer")

if (interactive()) deck

File: javascript/src/test-components/path-layer.js
const PATH_LAYER = {
  className: "PathLayer",
  properties: {
    pickable: true,
    widthScale: 20,
    widthMinPixels: 2,
    getPath: "@=path", //data => data.path,
    getColor: "@=color", // d => d.color, // [25,145, 12],
    getWidth: 5,
    tooltip: { html: "{{name}}" }
  },
  data: "https://raw.githubusercontent.com/uber-common/deck.gl-data/master/website/bart-lines.json"
};

export default {
  latitude: 37.8,
  longitude: -122.45,
  zoom: 10.5,
  pitch: 25,
  bearing: 0,
  layers: [
    PATH_LAYER
  ],
  calls: [
    { funcName: "addLayer", args: PATH_LAYER },
    { funcName: "addJSONEditor", args: {} }
  ]
};
